src/algebra/module/linear_map.lean

```diff
@@ -843,4 +843,18 @@ def to_module_End : S →+* module.End R M :=
   map_add' := λ f g, linear_map.ext $ add_smul _ _,
   ..distrib_mul_action.to_module_End R M }
 
+/-- The canonical (semi)ring isomorphism from `Rᵐᵒᵖ`
+to `module.End R R` induced by the right multiplication.
+-/
+def module_End_self : Rᵐᵒᵖ ≃+* module.End R R :=
+{ to_fun := λ a, ⟨λ x, x * mul_opposite.unop a, λ _ _, add_mul _ _ _, by simp only [mul_assoc,
```

Comment:
I think the forward direction might be `module.to_module_End`, which also gives you the `map_add` and `map_mull` proofs for free

Reply:
Thanks a lot! I was not aware that we can also use `distrib_mul_action.to_linear_map` for this case.